stytch/b2b/organizations/types.go

```diff
@@ -0,0 +1,463 @@
+//   - NextCursor: The `next_cursor` string is returned when your search result contains more than one page
+//     of results. This value is passed into your next search call in the `cursor` field.
+type ResultsMetadata struct {
+	Total      int32  `json:"total,omitempty"`
+	NextCursor string `json:"next_cursor,omitempty"`
+}
+
+// SSORegistration: Fields:
+//   - ConnectionID: Globally unique UUID that identifies a specific SSO `connection_id` for a Member.
+//   - ExternalID: The id given by the identity provider.
+//   - RegistrationID: The id that registers an SSO connection to a Member upon successful authentication.
+//   - SsoAttributes: An object for storing SSO attributes brought over from the identity provider.
+type SSORegistration struct {
+	ConnectionID   string         `json:"connection_id,omitempty"`
+	ExternalID     string         `json:"external_id,omitempty"`
+	RegistrationID string         `json:"registration_id,omitempty"`
+	SsoAttributes  map[string]any `json:"sso_attributes,omitempty"`
+}
+
+// SearchParams: Request type for `Organizations.Search`.
+// Fields:
+//   - Cursor: The `cursor` field allows you to paginate through your results. Each result array is limited
+//     to 1000 results. If your query returns more than 1000 results, you will need to paginate the responses
+//     using the `cursor`. If you receive a response that includes a non-null `next_cursor` in the
+//     `results_metadata` object, repeat the search call with the `next_cursor` value set to the `cursor` field
+//     to retrieve the next page of results. Continue to make search calls until the `next_cursor` in the
+//     response is null.
+//   - Limit: The number of search results to return per page. The default limit is 100. A maximum of 1000
+//     results can be returned by a single search request. If the total size of your result set is greater than
+//     one page size, you must paginate the response. See the `cursor` field.
+//   - Query: The optional query object contains the operator, i.e. `AND` or `OR`, and the operands that
+//     will filter your results. Only an operator is required. If you include no operands, no filtering will be
+//     applied. If you include no query object, it will return all Organizations with no filtering applied.
+type SearchParams struct {
+	Cursor string      `json:"cursor,omitempty"`
+	Limit  uint32      `json:"limit,omitempty"`
+	Query  SearchQuery `json:"query,omitempty"`
+}
+
+// SearchQuery: Fields:
+//
+//   - Operator: The action to perform on the operands. The accepted value are:
+//
+//     `AND` – all the operand values provided must match.
+//
+//     `OR` – the operator will return any matches to at least one of the operand values you supply.
+//
+//   - Operands: An array of operand objects that contains all of the filters and values to apply to your
+//     search query.
+type SearchQuery struct {
+	Operator string           `json:"operator,omitempty"`
```

Comment:
Would be nice to have Operator be its own type
```
type Operator string
const OperatorAnd = "And"
...
```

Reply:
Yeah... let me see if there's a way I can hard-code search queries specifically since we *know* what the valid values are.

Reply:
Yeah... let me see if there's a way I can hard-code search queries specifically since we know what the valid values are.

Reply:
Done now!